python_modules/dagster/dagster/_core/storage/event_log/sqlite/consolidated_sqlite_event_log.py

```diff
@@ -113,7 +113,9 @@ def index_connection(self):
 
     def has_table(self, table_name: str) -> bool:
         engine = create_engine(self._conn_string, poolclass=NullPool)
-        return bool(engine.dialect.has_table(engine.connect(), table_name))
+        with engine.connect() as conn:
+            has_table = bool(engine.dialect.has_table(conn, table_name))
+        return has_table
```

Comment:
do you see this getting called a lot? I know i added memoization to one of these table checks for concurrency because it was re-querying a ton

Reply:
It only gets called twice, but just saw that hanging connections might cause some timeouts so wanted to clean up.

Also, some wishful thinking on my part that this will resolve some of the test failures, which only happen to affect the `ConsolidatedSqliteEventLogStorage`